torch/_inductor/index_propagation.py

```diff
@@ -144,6 +144,9 @@ def __post_init__(self):
         ), "Symbolic IndexPropVar must contain a TypedExpr"
 
 
+IndexPropResult = Union[IndexPropVar, Tuple["IndexPropResult", ...]]
```

Comment:
```suggestion
IndexPropResult : TypeAlias = Union[IndexPropVar, Tuple["IndexPropResult", ...]]
```
import TypeAlias from typing_extensions